File: backend/domain/models/major.go
package models

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Major struct {
	ID                uuid.UUID  `gorm:"type:uuid;primaryKey"`
	Title             string     `gorm:"index"`
	School            string     `gorm:"not null"`
	CohortYear        int        `gorm:"index"`
	RequirementsBoxID *uuid.UUID `gorm:"type:uuid;index"`
	RequirementsBox   *Box       `gorm:"foreignKey:RequirementsBoxID"`
}

func (m *Major) BeforeCreate(tx *gorm.DB) error {
	if m.ID == uuid.Nil {
		m.ID = uuid.New()
	}
	return nil
}

File: src/Domain/Entities/Major.cs
namespace CuRoadmap.Domain.Entities;

public class Major
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string School { get; set; } = string.Empty;
    public int CohortYear { get; set; }
    public Guid? RequirementsBoxId { get; set; }

    public Box? RequirementsBox { get; set; }
    public ICollection<Specialization> Specializations { get; set; } = new List<Specialization>();
}
